WikiWatershed/model-my-watershed
# Upgrade development environment to PostgreSQL 9.6

## Overview

In addition to adding support for PostgreSQL 9.6, upgrade PostGIS to 2.3, which coincides with the version utilized by RDS in staging and production.

Lastly, the project specific init script for PostgreSQL was removed because it only existed to emit a `postgresql-started` event for the `pgweb` service configuration to subscribe to.

Fixes #3045 

## Testing Instructions

- To avoid mixing up state between the PostgreSQL 9.4 and 9.6 installations, I recreated all of the project virtual machines from scratch. The `app`, `tiler`, and `worker` virtual machines should be recreated in addition to the `services` virtual machine because they install PostgreSQL client libraries too.

```console
$ vagrant destroy -f
$ vagrant up
```

- Next, apply migrations and load some of the boundary data:

```console
$ ./scripts/manage.sh migrate
$ vagrant ssh app
vagrant@app:~$ cd /vagrant/scripts/aws/
vagrant@app:/vagrant/scripts/aws$ ./setupdb.sh ...
```

- Test the application, the tile service, and background jobs to ensure that database connectivity works
- Lastly, validate the versions of PostgreSQL and PostGIS

```
$ vagrant ssh services
$ sudo su postgres
postgres@services:/home/vagrant$ psql
psql (9.6.15)
Type "help" for help.

postgres=# \c mmw
You are now connected to database "mmw" as user "postgres".
mmw=# select version();
                                                                     version
-------------------------------------------------------------------------------------------------------------------------------------------------
 PostgreSQL 9.6.15 on x86_64-pc-linux-gnu (Ubuntu 9.6.15-1.pgdg16.04+1), compiled by gcc (Ubuntu 5.4.0-6ubuntu1~16.04.11) 5.4.0 20160609, 64-bit
(1 row)

mmw=# select postgis_version();
            postgis_version
---------------------------------------
 2.3 USE_GEOS=1 USE_PROJ=1 USE_STATS=1
(1 row)
```
